crates/autopilot/src/solvable_orders.rs

```diff
@@ -795,6 +824,21 @@ impl OrderFilterCounter {
         filtered_orders.into_keys().collect()
     }
 
+    /// Creates a new checkpoint based on the found invalid orders.
+    fn checkpoint_by_invalid_orders(&mut self, reason: Reason, invalid_orders: &[OrderUid]) {
+        *self.counts.entry(reason).or_default() += invalid_orders.len();
+        for order_uid in invalid_orders {
+            self.orders.remove(order_uid).unwrap();
```

Comment:
If an order gets removed for multiple reasons this `.unwrap()` would lead to a panic IIUC.
Let's just ignore the return value from `.remove()` here since we don't really care about the value anyway.
```suggestion
            let _ = self.orders.remove(order_uid);
```